extensions/sigstore.md

```diff
@@ -0,0 +1,51 @@
+# Signature Extension: Sigstore
+
+July 20, 2023
+
+Version: TBD
+
+Status: WIP
+
+This document describes the
+[signature extension](../envelope.md#signature-extensions) schema for the
+Sigstore ecosystem.
+
+## Definition
+
+The Sigstore extension's type must be set to `https://sigstore.dev`.
+
+The extension's fields are identical to the
+[`VerificationMaterial` schema](https://github.com/sigstore/protobuf-specs/blob/d1814679400076ca4bc24f54e54b918dda6d2bb2/protos/sigstore_bundle.proto#L52-L68)
+defined by Sigstore. Note: the X.509 certificate chain embedded in the extension
+MUST NOT include the root certificate.
+
+FIXME: we need to identify restrictions on the extension's type. Also, what is
+the policy on registering well known extensions? If DSSE points to an external
+definition, we have to ensure the link remains live and correct. On the other
```

Comment:
Do we have to?  Could we say "you're responsible for ensuring the link is live and correct"?

Maybe when folks submit the links to their specification they should be required to include the repo/bug tracker they use to manage the spec so they can be contacted in case things break?

It does seem similar to what in-toto attestations does (storing them in the repo).  That is a bit of a burden to be sure.

Reply:
My vote would be just storing links where possible. The full extension definition might change/have new versions, and it's nice to have that in just one place. The links might get out of date, but probably less often than the specification changes.

Reply:
This makes sense to me. Storing links is fine, for the sigstore example, can we add some mention of this to use the VerificationMaterials definition, @haydentherapper?

Reply:
I agree with links. Specification will hopefully come eventually, we can point to that later on.

One question is whether we should point to `main` or a specific commit. I think both come with downsides - main might contain breaking changes not yet reflected in a released version, but a specific commit might ignore the latest updates.

Reply:
If we have the version recorded here with a link to the sigstore doc, then `main` won't work. The version recorded here must map to the specific definition for that version on the sigstore side so I suspect we're going to have to use the commit (or tag).

Reply:
Since we want to do links only, I've renamed this folder and created an extensions.md file with a table to list extensions with links. Is that preferable?